server/cluster.go

```diff
@@ -99,10 +100,28 @@ func (s *clusterServer) DownloadCluster(ctx context.Context, cluster *infrapbv3.
 	}, nil
 }
 
-func (s *clusterServer) UpdateClusterStatus(ctx context.Context, cluster *infrapbv3.Cluster) (*infrapbv3.Cluster, error) {
-	err := s.UpdateClusterConditionStatus(ctx, cluster)
+func (s *clusterServer) UpdateClusterStatus(ctx context.Context, request *rpcv3.UpdateClusterStatusRequest) (*rpcv3.UpdateClusterStatusResponse, error) {
+	err := s.UpdateClusterConditionStatus(ctx, &infrav3.Cluster{
+		Metadata: request.Metadata,
+		Spec: &infrav3.ClusterSpec{
+			ClusterData: &infrav3.ClusterData{
+				ClusterStatus: request.ClusterStatus,
```

Comment:
To my understanding, this will override all cluster conditions? If true, then in [relay PR](https://github.com/paralus/relay/pull/108/files#diff-3be571c2df0fa983697d864c38051292db170972d025e6f05153246adf281991R48), we first need to fetch cluster conditions, make UpdateClusterStatus request after modifying **only** ClusterHealth condition to keep other conditions intact.

Reply:
No, we use a patch strategy in UpdateClusterStatus which takes care of updating only the given conditions.

Reply:
Alright. This makes sense then.